website/src/components/Demo.tsx

```diff
@@ -49,16 +49,36 @@ function useCounterContract() {
 }
 
 function DemoAccount() {
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(null)
   const { account, connect } = useStarknet()
+  const injected = useMemo(() => new InjectedConnector(), [])
+
+  useEffect(() => {
+    if (connect && injected) {
```

Comment:
This looks good. It should be moved to the starknet provider so that users can reuse it. My proposed APi is

```ts
<StarknetProvider autoConnect>
...
</StarknetProvider>